cmd/engflow_auth/main_test.go

```diff
@@ -343,8 +343,10 @@ func TestRun(t *testing.T) {
 			wantErr:  "expected exactly 1 positional argument",
 		},
 		{
-			desc: "logout with unknown cluster",
-			args: []string{"logout", "unknown.example.com"},
+			desc:     "logout with unknown cluster",
+			args:     []string{"logout", "unknown.example.com"},
+			wantCode: autherr.CodeBadParams,
+			wantErr:  "no credentials found for cluster",
```

Comment:
I'd like to see a couple test cases here cover the fallback mechanism, at least for `login` and `get`. Right now I think we don't test that the `-store` flag can be set.

Unit tests are nice to have, but integration tests are often better at finding bugs.

Reply:
I added `--store=file login cluster.example.com`, ~and can add some error cases (unknown store flag value)~ (this is also hard because the test skips these codepaths) - but the value of the tests is limited here, since the entire store backend is faked by the test, so `Fallback` (or any other token storage backend) is not in the test path. IOW, the test as written today isolates functionality in main itself, and aren't full integration tests - that would be a larger change.

Reply:
Let's leave it for now then, but I'm not really happy with this state. Generally we should have integration tests that cover the full app.